Assignment4/UniversityCourse/UniversityCourse.cs

```diff
@@ -0,0 +1,79 @@
+﻿using System;
+using System.Text.RegularExpressions;
+
+namespace Assignment4
+{
+    public class UniversityCourse : Event
+    {
+        public static int NumberOfCourses { get; private set; } = 0;
+
+        private string _CourseID;
+        public string CourseID
+        {
+            get { return _CourseID; }
+            set {
+                if (value is null)
+                    throw new ArgumentNullException("The course ID cannot be null", nameof(value));
+                if(!Regex.IsMatch(value, @"\w{4}\d{3}"))
+                    throw new ArgumentException($"The course ID must be 4 characters followed by 3 digits: {value}", nameof(value));
+                _CourseID = value;
+            }
+        }
+
+
+        private string _Instructor;
+        public string Instructor
+        {
+            get { return _Instructor; }
+            set {
+                if (value is null)
+                    throw new ArgumentNullException("The instructor name cannot be null", nameof(value));
+                value = value.Trim();
+                if (value.Length < 2)
+                    throw new ArgumentOutOfRangeException($"The instructor name must be atleast 2 characters: {value}", nameof(value));
```

Comment:
Typo. `atleast` to `at least`. You did this another time in this file, and two other times in Event.
```suggestion
                    throw new ArgumentOutOfRangeException($"The instructor name must be at least 2 characters: {value}", nameof(value));
```